File: library-ola/src/main/java/com/angelstar/ola/utils/ThreadUtils.java
package com.angelstar.ola.utils;

import android.os.Handler;
import android.os.Looper;

/**
 * @Author：yangbaojiang
 * @Date: 2022/10/26 14:57
 * @E-Mail: pgrammer.ybj@outlook.com
 * @Description: android
 */
public class ThreadUtils {
    public static final Handler uiHandler= new Handler(Looper.getMainLooper());
}

File: editor-core/src/main/java/com/ss/ugc/android/editor/core/utils/ThreadUtils.kt
package com.ss.ugc.android.editor.core.utils

import android.os.Handler
import android.os.Looper

/**
 * @since 17/8/21
 * @author joffychim
 */
private val uiHandler: Handler by lazyOf(Handler(Looper.getMainLooper()))

val isUiThread: Boolean
    get() = Looper.getMainLooper() == Looper.myLooper()

fun postOnUiThread(delayMs: Long = 0, block: () -> Unit) {
    if (delayMs > 0) {
        uiHandler.postDelayed({
            block()
        }, delayMs)
    } else {
        uiHandler.post {
            block()
        }
    }
}

fun runOnUiThread(delayMs: Long = 0, block: () -> Unit) {
    if (delayMs > 0) {
        uiHandler.postDelayed({
            block()
        }, delayMs)
    } else {
        if (isUiThread) {
            block()
        } else {
            uiHandler.post {
                block()
            }
        }
    }
}
